.github/workflows/main.yml

```diff
@@ -0,0 +1,123 @@
+name: CI
+
+# ensures that main has our basic test and code quality
+on:
+  push:
+    branches: [main]
```

Comment:
ensures that all merges in main will pass the pipeline again to get test coverage and ensure main has a complete pipeline